pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -458,10 +467,27 @@ def log(
                 f"You called `self.log({name}, ...)` twice in `{fx}` with different arguments. This is not allowed"
             )
 
-        if batch_size is not None:
-            self.batch_size = batch_size
+        # check if we have extracted the batch size already
+        if batch_size is None:
```

Comment:
Let's move this logic within its own private method

Reply:
done